xblock/fields.py

```diff
@@ -549,7 +558,7 @@ def __delete__(self, xblock):
         # Since we know that the field_data no longer contains the value, we can
         # avoid the possible database lookup that a future get() call would
         # entail by setting the cached value now to its default value.
-        self._set_cached_value(xblock, copy.deepcopy(self.default))
+        self._set_cached_value(xblock, self._get_default_value_to_cache(xblock))
```

Comment:
Hm. The intention was that `Sentinels` would be singletons, so they should probably behave correctly under deepcopy (and just return themselves, so that `is` checks work).